nqeit-iceberg-dremio-extension/build.gradle.kts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+  `java-library`
+  `tools-integrations-conventions`
+  id("org.projectnessie")
+}
+
+val versionRestAsssured = "5.3.0"
+
+dependencies {
+  compileOnly(platform(rootProject))
+
+  compileOnly(project(":nqeit-nessie-common"))
+  compileOnly(libs.findbugs.jsr305)
+  compileOnly(libs.microprofile.openapi)
+  compileOnly(libs.jetbrains.annotations)
+  compileOnly(libs.errorprone.annotations)
+
+  compileOnly(platform(libs.junit.bom))
+  compileOnly(libs.junit.jupiter.engine)
+  compileOnly(libs.bundles.junit.testing)
+
+  implementation("io.rest-assured:rest-assured:$versionRestAsssured")
```

Comment:
I will be using the Apache HttpComponents library. Is it fine to use this library? or if you could recommend some good library.

Reply:
The point is, that you're plumbing the requests manually and parse the responses manually as well. I'd really prefer to use generated code based on the OpenAPI schema.

Reply:
can you link to the OpenAPI schema?
because we can not find it on https://docs.dremio.com/cloud/api/

also wondering:
is it really necessary to generate a full blown client from it?
as is, we are only making ~3 distinct/simple http calls in `DremioHelper` and they are well isolated internally.
we can rework/enhance this at a later point in time if really required.

i'd argue it is "good enough" as is and working on automatically running these tests would be higher priority than reworking this well isolated internal detail.

Reply:
Not opposed to add it later. I assumed, that there is an OpenAPI schema available. Weird, that there are API examples, but not schema.

Reply:
Added instructions to README for getting the required properties and replaced the rest-assured library with http-client5.